core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -682,6 +690,48 @@ class ReplicaManager(val config: KafkaConfig,
             FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY)
         }
 
+        println("***************************Go inside read*******************************");
+
+        var batchs = logReadInfo.records.batches().iterator()
+        while(batchs.hasNext()){
+          var currentBatch = batchs.next();
+          var records = currentBatch.streamingIterator(decompressionBufferSupplier);
+          while(records.hasNext())
+          {
+            var record = records.next();
+            var valueByteArray = Utils.toArray(record.key());
+            for( a <- 0 to (valueByteArray.length - 1)) {
+              println("********************************** record.key() = %d ************************".format(valueByteArray(a)));
+            }
+          }
+        }
+
+        while (batchs.hasNext()) {
```

Comment:
Reset the batchs iterator first?

Reply:
yes, should be. But I test this two separately and add them together, so this is not the issue cause get out key failed.